projects/aca-content/folder-rules/src/rule-list/rule-list-item/rule-list-item.ui-component.spec.ts

```diff
@@ -0,0 +1,116 @@
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RuleListItemUiComponent } from './rule-list-item.ui-component';
+import { NoopTranslateModule, UnitTestingUtils } from '@alfresco/adf-core';
+import { Rule } from '../../model/rule.model';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { MatSlideToggleHarness } from '@angular/material/slide-toggle/testing';
+
+describe('RuleListItemUiComponent', () => {
+  let component: RuleListItemUiComponent;
+  let fixture: ComponentFixture<RuleListItemUiComponent>;
+  let loader: HarnessLoader;
+  let unitTestingUtils: UnitTestingUtils;
+
+  const mockRule = {
+    id: 'test-rule-id',
+    name: 'Test Rule',
+    description: 'Test rule description',
+    isEnabled: true
+  } as Rule;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [NoopTranslateModule, RuleListItemUiComponent]
+    });
+
+    fixture = TestBed.createComponent(RuleListItemUiComponent);
+    component = fixture.componentInstance;
+    loader = TestbedHarnessEnvironment.loader(fixture);
+    unitTestingUtils = new UnitTestingUtils(fixture.debugElement);
+    component.rule = mockRule;
+  });
+
+  it('should display rule name and description', () => {
+    fixture.detectChanges();
+
+    const nameElement = unitTestingUtils.getByCSS('.aca-rule-list-item__header__name');
+    const descriptionElement = unitTestingUtils.getByCSS('.aca-rule-list-item__description');
+
+    expect(nameElement.nativeElement.textContent.trim()).toBe(mockRule.name);
+    expect(descriptionElement.nativeElement.textContent.trim()).toBe(mockRule.description);
+  });
+
+  it('should show slide toggle when showEnabledToggle is true', async () => {
+    component.showEnabledToggle = true;
+    fixture.detectChanges();
+
+    const toggleHarness = await loader.getHarnessOrNull(MatSlideToggleHarness);
+    expect(toggleHarness).toBeTruthy();
+  });
+
+  it('should hide slide toggle when showEnabledToggle is false', async () => {
+    component.showEnabledToggle = false;
+    fixture.detectChanges();
+
+    const toggleHarness = await loader.getHarnessOrNull(MatSlideToggleHarness);
+    expect(toggleHarness).toBeFalsy();
+  });
+
+  it('should set toggle checked state based on rule.isEnabled', async () => {
+    component.showEnabledToggle = true;
+    component.rule.isEnabled = true;
+    fixture.detectChanges();
+
+    const toggleHarness = await loader.getHarness(MatSlideToggleHarness);
+    expect(await toggleHarness.isChecked()).toBe(true);
+  });
+
+  describe('onToggleClick', () => {
+    it('should stop event propagation and emit enabledChanged with isEnabled value', () => {
+      spyOn(component.enabledChanged, 'emit');
+      const mockEvent = jasmine.createSpyObj('Event', ['stopPropagation']);
```

Comment:
movkEvent is actually treated like any
<img width="860" height="66" alt="image" src="https://github.com/user-attachments/assets/b54f44eb-447f-48cd-aa24-5ba485fae055" />

 If you check createSpyObj you can see that it's generic function so you can do something like: 
```
jasmine.createSpyObj<Event>('Event', ['stopPropagation']);
```

After that mockEvent will have correct type